nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPackagesBuilder.java

```diff
@@ -70,9 +71,8 @@ public RPackagesBuilder(final String packagesPath) {
    * @param asset The asset to process.
    */
   public void append(final Asset asset) {
-    // is this asset at this particular path?
-    String base = basePath(packagesPath);
-    if (base.equals(basePath(asset.name()))) {
+    // is this asset at this particular path and archive kind?
+    if (packagesBasePath.equals(getBasePath(asset.name())) && extractAssetKind(asset) == ARCHIVE) {
```

Comment:
check is it possible to query archives only

Reply:
done in ed4f13fe223cdff2a4e750cca1cc41fbc1f95601